src/nspawn/nspawn.c

```diff
@@ -352,6 +352,15 @@ static int detect_unified_cgroup_hierarchy(void) {
         return 0;
 }
 
+static void parse_share_ns_env(const char *name, unsigned long mask) {
+        int r;
+
+        r = getenv_bool(name);
+        if (r < 0 && r != -ENXIO)
+                log_warning_errno(r, "Failed to parse %s from environment, defaulting to false.", name);
+        arg_share_ns = (arg_share_ns & ~mask) | (r > 0 ? 0 : mask);
```

Comment:
I can add an early-return for -ENXIO (no env var) as I agree it feels cleaner. At the moment it is *setting* the bit (false branch of the ternary operator sets the bit, which means ns will *not* be shared) but I see the confusion due to mis-naming in your other comment, so let's tackle both.